gtda/homology/simplicial.py

```diff
@@ -721,3 +722,272 @@ def plot(Xt, sample=0, homology_dimensions=None):
         """
         return plot_diagram(
             Xt[sample], homology_dimensions=homology_dimensions)
+
+
+@adapt_fit_transform_docs
+class FlagserPersistence(BaseEstimator, TransformerMixin, PlotterMixin):
+    """:ref:`Persistence diagrams <persistence_diagram>` resulting from
+    :ref:`filtrations <filtered_complex>` of :ref:`directed or undirected flag
+    complexes <clique_and_flag_complexes>`.
+
+    Given a weighted directed or undirected graph, information about the
+    appearance and disappearance of topological features (technically,
+    :ref:`homology classes <homology_and_cohomology>`) of various dimension
+    and at different scales is summarised in the corresponding persistence
+    diagram.
+
+    Parameters
+    ----------
+    homology_dimensions : list or tuple, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    directed : bool, optional, default: ``True``
+        If ``True``, :meth:`transform` computes the persistence diagrams of
+        filtered directed flag complexes arising from weighted directed
+        graphs. If False, computes the persistence diagrams of undirected
+        filtered flag complexes obtained by considering all weighted edges of
+        each weighted undirected graphs of the input collection as undirected,
+        and it is therefore sufficient (but not necessary) to pass a collection
+        of upper-triangular matrices. When ``False``, in each weighted
+        undirected graphs of the input collection, if two directed edges
+        corresponding to the same undirected edge are assigned different
+        weights, only the one on the upper triangular part of the adjacency
+        matrix is considered.
+
+    filtration : string, optional, default: ``'max'``
+        Algorithm determining the filtration. Warning: if an edge filtration is
+        specified, it is assumed that the resulting filtration is consistent,
+        meaning that the filtration value of every simplex of dimension at
+        least two should evaluate to a value that is at least the maximal value
+        of the filtration values of its containing edges. For performance
+        reasons, this is not checked automatically.  Possible values are:
+        ['dimension', 'zero', 'max', 'max3', 'max_plus_one', 'product', 'sum',
+        'pmean', 'pmoment', 'remove_edges', 'vertex_degree']
+
+    coeff : int prime, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    max_edge_weight : float, optional, default: ``numpy.inf``
+        Maximum edge weight to be considered in the filtration. All edge
+        weights greater than that value will be considered as
+        infinitely-valued, i.e., absent from the filtration and topological
```

Comment:
I think "infinitely-valued" can be a little confusing here because in fact if you play with `infinity_values` to change this. Do you agree?

Reply:
I will write some extra test and compare the meaning of `infinty_values` in VR and Flagser as per your main comment and come back to you.

Reply:
Thanks!

Reply:
Agreed, I simply removed "infintely-valued".

Reply:
Thanks!